src/transformers/models/depth_pro/modeling_depth_pro.py

```diff
@@ -0,0 +1,1676 @@
+            input_dims=config.hidden_size,
+            intermediate_dims=config.hidden_size,
+            output_dims=config.scaled_images_feature_dims[0],
+            n_upsample_layers=1,
+            use_proj=False,
+            bias=True,
+        )
+
+        # for STEP 7: fuse low_res and image features
+        self.fuse_image_with_low_res = nn.Conv2d(
+            in_channels=config.scaled_images_feature_dims[0] * 2,
+            out_channels=config.scaled_images_feature_dims[0],
+            kernel_size=1,
+            stride=1,
+            padding=0,
+            bias=True,
+        )
+
+    def forward(
+        self,
+        pixel_values: torch.Tensor,
+        head_mask: Optional[torch.Tensor] = None,
+        output_attentions: bool = False,
+        output_hidden_states: bool = False,
+        return_dict: bool = True,
+    ) -> Union[tuple, DepthProOutput]:
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if pixel_values.dim() != 4:
+            raise ValueError("Input tensor must have shape (B, C, H, W).")
+
+        B, C, H, W = pixel_values.shape
+
+        if not (C == self.config.num_channels):
+            raise ValueError(
+                f"Found {C} channels in image, expected number of channels is {self.config.num_channels} from config."
+            )
+
+        # pixel_values.shape (B, config.num_channels, config.image_size, config.image_size)
+
+        # STEP 1: create 3-level image
+
+        scaled_images = []
+        for ratio in self.scaled_images_ratios:
+            scaled_images.append(interpolate(pixel_values, ratio))
+            # (B, config.num_channels, config.image_size * ratio, config.image_size * ratio)
+
+        # STEP 2: create patches
+
+        for i in range(self.n_scaled_images):
+            scaled_images[i] = patch(
+                scaled_images[i],
+                patch_size=self.config.patch_size,
+                overlap_ratio=self.scaled_images_overlap_ratios[i],
+            )
+        scaled_images_num_patches = [len(i) for i in scaled_images]
+        patches = torch.cat(scaled_images[::-1], dim=0)  # -1 as patch encoder expects high res patches first
+        # (sum(scaled_images_num_patches), config.num_channels, config.patch_size, config.patch_size)
+
+        # STEP 3: apply patch and image encoder
+
+        patch_encodings = self.patch_encoder(
+            patches,
+            head_mask=head_mask,
+            output_attentions=output_attentions,
+            # required for intermediate features
+            output_hidden_states=self.n_intermediate_hooks or output_hidden_states,
+            return_dict=True,
```

Comment:
@qubvel  I don't understand, can you explain what is meant by the other option?

Reply:
I mean each module should have both options enabled for `return_dict`, we should avoid hardcoding  `return_dict=True`

Reply:
yes we do have both options available
I am using hard coded `return_dict=True` so that I can use it in the module accordingly but when returning, I am considering `return_dict` specified by the user
```py
        patch_encodings = self.patch_encoder(
            ...
            return_dict=True,
        )
        ...
        ...
        if not return_dict:
            return tuple(v for v in [last_hidden_state, features, hidden_states, attentions] if v is not None)

        return DepthProOutput(
            ...
        )
```
So the user still gets the both options